app/Platform/Actions/UpdateGameAchievementsMetrics.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Platform\Enums\AchievementFlag;
+use App\Platform\Models\Achievement;
+use App\Platform\Models\Game;
+
+class UpdateGameAchievementsMetrics
+{
+    public function execute(Game $game): void
+    {
+        // TODO refactor to do this for each achievement set
+
+        $gameId = (int) $game->id;
+
+        $parentGameId = getParentGameIdFromGameId($gameId);
+
+        // TODO refactor to player_achievements
+
+        $achievementsData = legacyDbFetchAll("SELECT ach.ID, ach.Points, SUM(CASE WHEN NOT ua.Untracked THEN aw.HardcoreMode ELSE 0 END) AS UnlocksHardcore
+              FROM Achievements AS ach
+              LEFT JOIN Awarded AS aw ON aw.AchievementID = ach.ID
+              LEFT JOIN UserAccounts AS ua ON ua.User = aw.User
+              WHERE ach.GameID = $game->id
+              AND ach.Flags = " . AchievementFlag::OfficialCore . "
+              GROUP BY ach.ID");
```

Comment:
This includes the softcore unlocks and also filters out Untracked users:

```php
$achievementsData = PlayerAchievement::select([
    'Achievements.ID',
    'Achievements.Points',
    DB::raw('COUNT(player_achievements.id) AS Unlocks'),
    DB::raw('SUM(CASE WHEN player_achievements.unlocked_hardcore_at IS NOT NULL THEN 1 ELSE 0 END) AS UnlocksHardcore'),
])
    ->leftJoin('Achievements', 'player_achievements.achievement_id', '=', 'Achievements.ID')
    ->leftJoin('UserAccounts', 'player_achievements.user_id', '=', 'UserAccounts.ID')
    ->where('Achievements.GameID', $gameId)
    ->where('Achievements.Flags', AchievementFlag::OfficialCore)
    ->where('UserAccounts.Untracked', 0)
    ->groupBy('Achievements.ID')
    ->get();
```

In the loop, the only thing that needs to be cast is `$unlocksHardcore`.

```php
$achievementId = $achievementData['ID'];
$achievementPoints = $achievementData['Points'];
$unlocks = $achievementData['Unlocks'];
$unlocksHardcore = intval($achievementData['UnlocksHardcore']);
```

Reply:
Thanks! I went a slightly different route to make it easier to reason about: https://github.com/RetroAchievements/RAWeb/pull/1874/commits/2ee360713a517a5e84962f897b8305bc67ba5e9c

Note: Casing using `(int)` is more efficient than `intval()`.